addons/block_code/ui/blocks/utilities/parameter_input/parameter_input.gd

```diff
@@ -56,6 +56,22 @@ var _drag_start: Vector2 = Vector2.INF
 	_v3_z_line_edit: "",
 }
 
+## Whether the text inputs can be edited, and whether the options selection have any effect.
+@export var editable: bool = true:
+	set = _set_editable
+
+
+func _set_editable(value) -> void:
+	if not is_node_ready():
+		await ready
+	editable = value
+	_line_edit.editable = value
+	_x_line_edit.editable = value
+	_y_line_edit.editable = value
+	_v3_x_line_edit.editable = value
+	_v3_y_line_edit.editable = value
+	_v3_z_line_edit.editable = value
```

Comment:
You could do this:

```suggestion
	for line_edit in _last_submitted_text:
		line_edit.editable = value
```

though I don't know if it's an accident that these 6 controls happen to match that dictionary.